watertap/property_models/activated_sludge/new_modified_asm2d_properties.py

```diff
@@ -0,0 +1,351 @@
+
+"""
+
+# Import Pyomo libraries
+import pyomo.environ as pyo
+
+# Import IDAES cores
+from idaes.core import (
+    declare_process_block_class,
+    MaterialFlowBasis,
+    PhysicalParameterBlock,
+    StateBlockData,
+    StateBlock,
+    MaterialBalanceType,
+    EnergyBalanceType,
+    LiquidPhase,
+    Solute,
+    Solvent,
+)
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.core.util.initialization import fix_state_vars, revert_state_vars
+import idaes.logger as idaeslog
+
+# Some more information about this module
+__author__ = "Marcus Holly"
+
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+
+@declare_process_block_class("NewASM2dParameterBlock")
+class NewASM2dParameterData(PhysicalParameterBlock):
+    """
+    Property Parameter Block Class
+    """
+
+    def build(self):
+        """
+        Callable method for Block construction.
+        """
+        super().build()
+
+        self._state_block_class = NewASM2dStateBlock
+
+        # Add Phase objects
+        self.Liq = LiquidPhase()
+
+        # Add Component objects
+        self.H2O = Solvent()
+
+        # Soluble species
+        self.S_A = Solute(
+            doc="Fermentation products, considered to be acetate. [kg COD/m^3]"
+        )
+        self.S_F = Solute(
+            doc="Fermentable, readily bio-degradable organic substrates. [kg COD/m^3]"
+        )
+        self.S_I = Solute(doc="Inert soluble organic material. [kg COD/m^3]")
+        self.S_N2 = Solute(
+            doc="Dinitrogen, N2. SN2 is assumed to be the only nitrogenous product of denitrification. [kg N/m^3]"
+        )
+        self.S_NH4 = Solute(doc="Ammonium plus ammonia nitrogen. [kg N/m^3]")
+        self.S_NO3 = Solute(
+            doc="Nitrate plus nitrite nitrogen (N03' + N02' -N). SN03 is assumed to include nitrate as well as nitrite nitrogen. [kg N/m^3]"
+        )
+        self.S_O2 = Solute(doc="Dissolved oxygen. [kg O2/m^3]")
+        self.S_PO4 = Solute(
+            doc="Inorganic soluble phosphorus, primarily ortho-phosphates. [kg P/m^3]"
+        )
+        self.S_K = Solute(doc="Potassium, [kg K/m^3]")
+        self.S_Mg = Solute(doc="Magnesium, [kg Mg/m^3]")
+        self.S_IC = Solute(doc="Inorganic carbon, [kg C/m^3]")
+
+        # Particulate species
+        self.X_AUT = Solute(doc="Autotrophic nitrifying organisms. [kg COD/m^3]")
+        self.X_H = Solute(doc="Heterotrophic organisms. [kg COD/m^3]")
+        self.X_I = Solute(doc="Inert particulate organic material. [kg COD/m^3]")
+        self.X_PAO = Solute(doc="Phosphate-accumulating organisms. [kg COD/m^3]")
+        self.X_PHA = Solute(
+            doc="A cell internal storage product of phosphorus-accumulating organisms, primarily comprising poly-hydroxy-alkanoates (PHA). [kg COD/m^3]"
+        )
+        self.X_PP = Solute(doc="Poly-phosphate. [kg P/m^3]")
+        self.X_S = Solute(doc="Slowly biodegradable substrates. [kg COD/m^3]")
```

Comment:
As a sanity check- do we expect this modification to comprise 18 components (excluding H2O)?

Reply:
Yep, base ASM2d has 19 components, but we removed 4 (X_TSS, X_MeOH, X_MeP, S_ALK) and added 3 (S_K, S_Mg, S_IC). These components are the first 18 listed in the Gujer matrix, and as we discussed, other components such as sulfur and iron can be added later.